poulpe-model/src/test/java/org/jtalks/poulpe/model/entity/PoulpeUserTest.java

```diff
@@ -31,4 +41,39 @@ public Group groupWithId(long id) {
         return group;
     }
 
+    @Test
+    public void theEntityFieldsShouldBeSerialized() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        PoulpeUser user = new PoulpeUser("username","email","password","salt");
+        user.setId(1);
+        user.setAvatar(new byte[]{1});
+        user.setVersion(1L);
+        user.setFirstName("First Name");
+        user.setLastName("Last Name");
+        user.setBanReason("Ban Reason");
+        user.setEnabled(true);
+        user.setRole("Role");
+        Method setLastLogin = User.class.getDeclaredMethod("setLastLogin", DateTime.class);
+        Method setEncodedUsername = User.class.getDeclaredMethod("setEncodedUsername", String.class);
+        setLastLogin.setAccessible(true);
+        setEncodedUsername.setAccessible(true);
+        setLastLogin.invoke(user,DateTime.now());
+        setEncodedUsername.invoke(user,"Encoded Username");
+
+        byte[] serialize = SerializationUtils.serialize(user);
+        PoulpeUser serializedUser = (PoulpeUser)SerializationUtils.deserialize(serialize);
+        assertReflectionEquals(user, serializedUser);
+    }
+
+    @Test
+    public void groupsFieldIsNotSerializable(){
+        List<Group> groups = PoulpeGroup.createGroupsWithNames("The Group");
+        List<Group> serializedGroups = new ArrayList<Group>();
+        try {
+            byte[] serialize = SerializationUtils.serialize(groups);
+            serializedGroups = (List<Group>) SerializationUtils.deserialize(serialize);
+        } catch (Exception e){
```

Comment:
Exception happens cause we directly and forcibly try to serialize non-serializable object.
AFAIU, when we provide default mechanim of serialization such non-serializable fields does not take part in the serialization (even these are non-transient and non-static) - as if these did not exsist at all.
Unfortunately, I can't prove this with [documentation](http://docs.oracle.com/javase/7/docs/platform/serialization/spec/serial-arch.html#5253), but it's easy to check with a simple code:
```java
public class Main implements Serializable {
    private static final long serialVersionUID = 1L;
    Object o = new Object();

    private void writeObject(ObjectOutputStream s) throws IOException {
        s.defaultWriteObject();
    }

    private void readObject(ObjectInputStream s) throws IOException, ClassNotFoundException {
        s.defaultReadObject();
    }

    public static void main(String[] args) {
    }
}
```
 During executing of this one no exceptions will be thrown.